### ISSUE TITLE
Custom Layouts don't work when using endpoints to populate data for template

### ISSUE THREAD

**Issue by hartwm** [sveltejs/kit#5595]
### Describe the bug

You can't use custom layouts if you use a named endpoint to separate data logic from template logic. 

### Reproduction

Create a structure such as 
`src/routes/hello.ts`
`src/routes/hello@foo.svelte`
`src/routes/__layout-foo.svelte`

GET request to `/hello`and it returns body data from hello.ts not html from hello@foo.svelte

### Logs

```shell
No console errors occur
```

### System Info

```shell
System:
    OS: macOS 12.2.1
    CPU: (10) x64 Apple M1 Pro
    Memory: 22.69 MB / 32.00 GB
    Shell: 5.1.4 - /usr/local/bin/bash
  Binaries:
    Node: 16.15.1 - ~/.nvm/versions/node/v16.15.1/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 8.11.0 - ~/.nvm/versions/node/v16.15.1/bin/npm
  Browsers:
    Brave Browser: 103.1.40.105
    Chrome: 103.0.5060.114
    Edge: 103.0.1264.62
    Firefox: 102.0.1
    Safari: 15.3
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.55 
    @sveltejs/adapter-vercel: ^1.0.0-next.48 => 1.0.0-next.60 
    @sveltejs/kit: next => 1.0.0-next.368 
    svelte: ^3.46.0 => 3.47.0 
    vite: ^2.9.13 => 2.9.14
```

### Severity

serious, but I can work around it

### Additional Information

I can work around it by combining data requests and template markup into a single svelte file

**Comment by Rich-Harris** [comment]
You can't have separate `hello.ts` and `hello@foo.svelte` routes — the endpoint should be renamed to `hello@foo.ts`. (The duplication is unfortunate, and would be solved by https://github.com/sveltejs/kit/discussions/5037, but there's some consensus-building to do before we get there.)

We should probably throw an error when you have conflicting routes like these.

**Comment by madeleineostoja** [comment]
Fixed by #5778
